pages/posts.js

```diff
@@ -0,0 +1,42 @@
+import getPosts from '../src/Services/get-posts';
+import Form from '../src/Components/Form/Form';
+import AddFormButton from '../src/Components/Form/AddFormButton';
+import { saveToLocal, loadFromLocal } from '../src/Components/lib/localStorage';
+import { useState, useEffect } from 'react';
+import dynamic from 'next/dynamic';
+
+export function getStaticProps() {
+	const data = getPosts();
+
+	return {
+		props: {
+			data,
+		},
+	};
+}
+export default function Posts({ data }) {
+	const [addPost, setAddPost] = useState(loadFromLocal('localPosts') ?? data);
+	const [formButton, setFormButton] = useState(false);
+
+	const RenderPosts = dynamic(() => import('../src/Components/Form/RenderPosts'), {
+		ssr: false,
+	});
+
+	useEffect(() => {
+		saveToLocal('localPosts', addPost);
+	}, [addPost]);
+
+	return (
+		<main>
+			<AddFormButton onSetFormButton={setFormButton} onFormButton={formButton} />
```

Comment:
I'd recommend to call your prop `formButton`(or `isFormActive` if you decide to rename it) when you pass the value of your `formButton` state to your `AddFormButton` component. `onFormButton` indicates that the value of your prop is a function which it is not in this case. 

```suggestion
			<AddFormButton onSetFormButton={setFormButton} formButton={formButton} />
```